docs/bytecode.md

```diff
@@ -0,0 +1,202 @@
+  outputSpatialDimensions: svarint[]
+}
+
+DotDimensionNumbersAttr {
+  lhsBatchingDimensions: svarint[]
+  rhsBatchingDimensions: svarint[]
+  lhsContractingDimensions: svarint[]
+  rhsContractingDimensions: svarint[]
+}
+
+FftTypeAttr
+  value: varint (encoded enum)
+}
+
+GatherDimensionNumbersAttr {
+  offsetDims: svarint[]
+  collapsedSliceDims: svarint[]
+  startIndexMap: svarint[]
+  indexVectorDim: svarint
+}
+
+PrecisionAttr {
+  value: varint (encoded enum)
+}
+
+RngAlgorithmAttr {
+  value: varint (encoded enum)
+}
+
+RngDistributionAttr {
+  value: varint (encoded enum)
+}
+
+ScatterDimensionNumbersAttr {
+  updateWindowDims: svarint[]
+  insertedWindowDims: svarint[]
+  scatterDimsToOperandDims: svarint[]
+  indexVectorDim: svarint
+}
+
+TransposeAttr {
+  value: varint (encoded enum)
+}
+
+TypeExtensionsAttr {
+  bounds : svarint[]
+}
+```
+
+### Types
+
+```
+TokenType {
+}
+```
+
+### Not Included:
+The following attributes / types are subclasses of builtin machinery and call
+into the bytecode implementations in the Builtin Dialect.
+
+- `StableHLO_ArrayOfLayoutAttr`
+- `StableHLO_BoolElementsAttr`
+- `StableHLO_FlatSymbolRefArrayAttr`
+- `StableHLO_LayoutAttr`
+- `HLO_ComplexTensor`
+- `HLO_Complex`
+- `HLO_DimensionTensor`
+- `HLO_DimensionValue`
+- `HLO_Float32Or64`
+- `HLO_Float`
+- `HLO_Fp32Or64Tensor`
+- `HLO_FpOrComplexTensor`
+- `HLO_FpTensor`
+- `HLO_IntFpOrComplexTensor`
+- `HLO_IntOrFpTensor`
+- `HLO_IntTensor`
+- `HLO_Int`
+- `HLO_PredIntOrFpTensor`
+- `HLO_PredOrIntTensor`
+- `HLO_PredTensor`
+- `HLO_Pred`
+- `HLO_QuantizedIntTensor`
+- `HLO_QuantizedInt`
+- `HLO_QuantizedSignedInt`
+- `HLO_QuantizedUnsignedInt`
+- `HLO_SInt`
+- `HLO_ScalarIntTensor`
+- `HLO_StaticShapeTensor`
+- `HLO_TensorOrTokenOrTuple`
+- `HLO_TensorOrToken`
+- `HLO_Tensor`
+- `HLO_Tuple`
+- `HLO_UInt`
+
+Additionally, despite its name, `StableHLO_ConvolutionAttributes` is not 
+an attribute and is not encoded. Rather, it is a dag which gets expanded
+into several attributes which are all encoded separately.
+
+### Still to do:
+
+The following attributes / types are not yet implemented:
+
+- CHLO_ComparisonDirectionAttr
+  + CHLO bytecode will come in a future changelist.
+- CHLO_ComparisonTypeAttr
+  + CHLO bytecode will come in a future changelist.
+
+## Other Notes
+
+### Testing Bytecode with Round Trips
+Testing that the round-trip of an MLIR file produces the same results is a good
+way to test that the bytecode is implemented properly.
+
+```
+$ stablehlo-opt -emit-bytecode stablehlo/tests/print_stablehlo.mlir | stablehlo-opt
+```
+
+### Find out what attributes or types are not encoded:
+Since attributes and types that don't get encoded are instead stored as strings,
+the `strings` command can be used to see what attributes were missed:
+
+_Note: The following trace is from a previous revision where the `scatter` attribute 
+was not implemented. Currently all types/attrs are implemented and log only shows 
```

Comment:
How about we remove this sentence and its accompanying output in the printout below? "The following trace is from a previous revision where the `scatter` attribute was not implemented". Even without `stablehlo.scatter`, the output is educational enough?